CONTRIBUTING.md

```diff
@@ -1,46 +1,56 @@
+* Claim your issue by commenting your intent to work on it to avoid duplication of efforts. 
+* Fork the repository on GitHub
+* Create a branch from where you want to base your work (usually master).
+* Make your changes. If you are working on code contributions, please see [Setting up the Development Environment](#setting-up-your-development-environment).
+* Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments) and the _Formatting and style_ section of Peter Bourgon's [Go: Best Practices for Production Environments](http://peter.bourgon.org/go-in-production/#formatting-and-style).
+* Commit your changes by making sure the commit messages convey the need and notes about the commit.
+* Push your changes to the branch in your fork of the repository.
+* Submit a pull request to the original repository. See [Pull Request checklist](#pull-request-checklist)
 
-You will need an Linux host with Vagrant 1.9.1+ and Virtual Box 5.1. Follow these steps to setup your Development Environment.
 
-```
-cd <your-work-dir-on-linux-host>
-git clone https://github.com/openebs/maya.git
-vagrant up master-01
-vagrant ssh master-01
-#Only for the first time
-make bootstrap
-make deps
-make dev
-```
+## Pull Request Checklist
+* Rebase to the current master branch before submitting your pull request.
+* Commits should be as small as possible. Each commit should follow these checklist:
+  - For code-changes, add tests relevant to the fixed bug or new feature. 
+  - Pass the compile and tests - includes spell checks, formatting.
+  - Commit header (first line) should convey what changed
+  - Commit body should include the details like why the changes is required and how the change has been made
+  - DCO Signed 
+* If your PR is not getting reviewed or you need a specific person to review it, please reach out to the OpenEBS Contributors. See [OpenEBS Community](https://github.com/openebs/openebs/tree/master/community)
 
+## Sign your work
 
-### Sign your work
-
-We use the Developer Certificate of Origin (DCO) as a additional safeguard
-for the OpenEBS project. This is a well established and widely used
-mechanism to assure contributors have confirmed their right to license
-their contribution under the project's license.
-Please read [dcofile](https://github.com/openebs/openebs/blob/master/contribute/developer-certificate-of-origin).
-If you can certify it, then just add a line to every git commit message:
+We use the Developer Certificate of Origin (DCO) as a additional safeguard for the OpenEBS project. This is a well established and widely used mechanism to assure contributors have confirmed their right to license their contribution under the project's license. Please read [dcofile](https://github.com/openebs/openebs/blob/master/contribute/developer-certificate-of-origin). If you can certify it, then just add a line to every git commit message:
```

Comment:
as an additional security (instead of safeguard?)
to assure contributors = to assure that contributors...

Reply:
done